test/functional/config.js

```diff
@@ -0,0 +1,29 @@
+const { readFileSync } = require('fs');
+
+let privateKey;
+if (process.env.TKPK) {
+  privateKey = readFileSync(process.env.TKPK, 'utf-8');
+}
+
+const config = {
+  transport: process.env.TKTRANSPORT || 'ssh',
+  transportOptions: {
+    database: process.env.TKDB || '*LOCAL',
+    username: process.env.TKUSER || '',
+    password: process.env.TKPASS || '',
+    host: process.env.TKHOST || 'localhost',
+    port: process.env.TKPORT,
+    path: process.env.TKPATH || '/cgi-bin/xmlcgi.pgm',
+    privateKey,
+    passphrase: process.env.TKPHRASE,
+    verbose: !!process.env.TKVERBOSE,
+    dsn: process.env.TKDSN,
+  },
+  restOptions: {
+    host: process.env.TKHOST || 'localhost',
+    port: process.env.TKPORT || 80,
+    path: process.env.TKPATH || '/cgi-bin/xmlcgi.pgm',
+  },
```

Comment:
Can all these be set or should we set the correct attributes based on TKTRANSPORT?

Reply:
They can all be set because using `Connection` constructor the `transportOptions` object will be passed directly to the transport. 

The transport will only use the properties it needs to run.

For example the ssh transport will not read the `dsn` key.

Reply:
In the tests/functional/deprecated/*.js tests `iConn` is used for connections. 

The signature is:

`iConn(database, username, password, restOptions)`

iConn determines to use the rest transport based on if the fourth parameter is an object.

That is why I have the `restOptions` key above.

We can remove `restOptions` in config.js and just create the restOptions object if the rest transport is used.

Instead of this 

```js

...

const { config } = require('../config');

let restOptions = null;
if (config.transport === 'rest') {
  restOptions = config.restOptions;
}

...

 const connection = new iConn(database, username, password, restOptions);
```

We can have

```js

...

const { config } = require('../config');

let restOptions = null;
if (config.transport === 'rest') {
  restOptions = {
    host: config.host,
    port: config.port,
    path: config.path,
  }
}

...


 const connection = new iConn(database, username, password, restOptions);
```

Reply:
Removed `restOptions` from the config object.